docs/generate_docs.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# script for generating documentation for NMODL
+
+set -xeu
+
+# in order to create the docs, we first need to build NMODL in some dir
+build_dir="${1:-"$(mktemp -d)"}"
+if ! [ -d "${build_dir}" ]
+then
+    mkdir -p "${build_dir}"
+fi
+
+# the Python executable to use (default: whatever `python3` is)
+python_exe="${2:-"$(command -v python3)"}"
+
+# should we use a separate venv?
+use_venv="${3:-}"
+
+if [ "${use_venv}" != "false" ]
+then
+    venv_name="$(mktemp -d)"
+    $(command -v python3) -m venv "${venv_name}"
+    . "${venv_name}/bin/activate"
+    python_exe="$(command -v python)"
+    ${python_exe} -m pip install -U pip
+else
+    python_exe="$(command -v python3)"
+fi
+${python_exe} -m pip install ".[docs]" -C build-dir="$(cd "${build_dir}"; pwd -P)"
+
+# the abs dir where this script is located (so we can call it from wherever)
+script_dir="$(cd "$(dirname "$0")"; pwd -P)"
+
+cd "${script_dir}"
+doxygen Doxyfile
+sphinx-build . "${script_dir}/../public"
+cd -
+
+if [ "${use_venv}" != "false" ]
+then
+    deactivate
```

Comment:
We leave the temporary directory in place, creating a new one each time we build the docs. If it's temporary, maybe we should delete it?

Reply:
You mean the `venv` dir? I prefer to leave the cleanup of those up to the OS, no need to run `rm -fr` unless absolutely necessary (while I quote my variables judiciously, I'm always paranoid of something like [this](https://github.com/MrMEEE/bumblebee-Old-and-abbandoned/issues/123) happening).

Reply:
Okay, then remove the `mktemp` and require users to pass in a valid build directory. In CI you can use `$(mktemp -d)` or something else.

The reasons is that starting from nothing each time is unacceptably slow (multiple minutes). Meaning the default should reuse an existing build directory. (Or create one the first time it runs.)

Also it doesn't tell where to find the docs:
```
generating indices... genindex py-modindex done
copying notebooks ... [100%] notebooks/nmodl-sympy-solver-sparse.ipynb                                                                                        
highlighting module code... [100%] nmodl.ode                                                                                                                  
writing additional pages... search done
copying images... [100%] images/ion_storage-opt.svg                                                                                                           
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 13 warnings.

The HTML pages are in ../public.
+ cd -
/home/lucg/sandbox/nmodl/pr-1147/nmodl
+ '[' '' '!=' false ']'
+ deactivate
+ '[' -n /usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/opt/cuda/nsight_compute:/opt/cuda/nsight_systems/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/lucg/.local/bin:/home/lucg/third_party/gmsh-4.8.3/bin:/opt/Xilinx/Vivado/2020.2/bin:/home/lucg/.local/share/gem/ruby/2.7.0/bin ']'
+ PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/opt/cuda/nsight_compute:/opt/cuda/nsight_systems/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/lucg/.local/bin:/home/lucg/third_party/gmsh-4.8.3/bin:/opt/Xilinx/Vivado/2020.2/bin:/home/lucg/.local/share/gem/ruby/2.7.0/bin
+ export PATH
+ unset _OLD_VIRTUAL_PATH
+ '[' -n '' ']'
+ '[' -n /usr/bin/bash -o -n '' ']'
+ hash -r
+ '[' -n '' ']'
+ unset VIRTUAL_ENV
+ unset VIRTUAL_ENV_PROMPT
+ '[' '!' '' = nondestructive ']'
+ unset -f deactivate
```
are the last couple of times of output. The location: `../public` is definitely wrong.

Reply:
I'm not sure why it's very careful about not choosing a build directory, but boldly claims `public` as it's own. One should also check that if there already content in `public` from a prior build that it'll be replaced correctly. A typical issue is when a file is removed from the documentation, it might still be present in `public`.

Would it make sense to move both the build dir and the directory `public` into the output directory which one passes to `generate_docs.sh`? That way if I want to delete everything I can simply `rm -r OUTPUT_DIR` and all generated content is gone, including the `venv`.

Reply:
Updated, docs now build in `[DIRECTORY]/docs`, where `[DIRECTORY]` is whatever dir you want, while the temporary build files are in `[DIRECTORY]/build` (see updated docs for usage).
On my machine, if re-building the docs, `bash docs/generate_docs.sh mydocs python false` runs in < 10 seconds.

Reply:
Doesn't work:
```
$ docs/generate_docs.sh build-docs python3
/home/lucg/sandbox/nmodl/pr-1147/nmodl
Running Sphinx v5.3.0
[NMODL] [warning] :: Python bindings are not available

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/tmp/tmp.2no3mfNQJo/lib/python3.11/site-packages/sphinx/config.py", line 350, in eval_config_file
    exec(code, namespace)
  File "/home/lucg/sandbox/nmodl/pr-1147/nmodl/docs/conf.py", line 83, in <module>
    version = nmodl.__version__
              ^^^^^^^^^^^^^^^^^
AttributeError: module 'nmodl' has no attribute '__version__'
```

Reply:
It seems to build docs and after installing via `pip install .` it seems like it may have created a functional module (import and printing `nmodl.__version__` don't fail).